packages/loopring_v3/contracts/impl/BasicDepositContract.sol

```diff
@@ -110,12 +120,21 @@ contract BasicDepositContract is IDepositContract, ReentrancyGuard
         override
         payable
         onlyExchange
+        nonReentrant
+        ifNotZero(amount)
+        returns (uint actualAmount)
     {
-        // Keep track how many tokens are deposited in the exchange
-        exchangeBalance[token] = exchangeBalance[token].sub(uint(amount));
-
-        // Transfer the tokens from the contract to the recipient
-        transferOut(to, token, amount);
+        actualAmount = amount;
+        if (isETHInternal(token)) {
+            to.sendETHAndVerify(actualAmount, gasleft());
+        } else {
+            try ERC20(token).transfer(to, actualAmount) {} catch {
```

Comment:
This may not work correctly for those "bad ERC20 tokens` that don't follow the standard correctly (like not throwing and only returning a boolean or things like that). I think it's safer to use `safeTransfer` here instead which will handle all those special cases.

Reply:
safeTransfer is not working, not an external function :(

Reply:
I figured it out.

Reply:
`safeTransfer` returns a boolean you can use, it does the try/catch internally.